### DISCUSSION TITLE
[Usage question] - How to retrieve session with particular dataset name (not type)

### DISCUSSION THREAD

**Discussion by GaelleChapuis** [int-brain-lab/iblenv#220] [Q&A] [answered]
**Describe the question you have**
I would like to retrieve sessions that have a particular dataset *name* (not type). For example, I would like to query sessions that have the specific datasets `['_ibl_bodyCamera.times', '_ibl_rightCamera.times']`

As example: this query on the dataset types works, but is not precise enough for my use case (as there is e.g. 3 camera files englobed into 1 dataset type)
```
# Check on dataset types (not precise enough for this case)
dsets = ['camera.times']
sess = one.alyx.rest('sessions', 'list',
                     django=f'data_dataset_session_related__dataset_type__name__in,{dsets}')
```

I was hoping to use a query using the filter `session__dataset__name__in`, but it does not seem to work from ONE:
```
# Check on specific dataset
dsets = ['_ibl_bodyCamera.times']
sess = one.alyx.rest('sessions', 'list',
                     django=f'data_dataset_session_related__dataset__name__in,{dsets}')
```
error : `"Related Field got invalid lookup: dataset"`

Could anyone point me to the correct filter to use?
Thanks a million !

**Is your question related to a specific product?**
ONE

**Comment by rossant** [comment, marked as answer]
does this work? `data_dataset_session_related__name__in`

**Comment by GaelleChapuis** [comment]
Yes this works!
```
dsets = ['_ibl_wheelMoves.intervals.npy']
sess = one.alyx.rest('sessions', 'list', django=f'data_dataset_session_related__name__in,{dsets}')
```

**Comment by GaelleChapuis** [comment]
Just a note on chaining queries:

```
dsets = ['_ibl_rightCamera.times.npy']
sess = one.alyx.rest('sessions', 'list', django=f'data_dataset_session_related__name__in,{dsets}')
```
>> returns over 700 sessions (mostly ephys)

When including more datasets in the query:
```
dsets = ['_ibl_rightCamera.times.npy', '_ibl_bodyCamera.times.npy', '_ibl_leftCamera.times.npy']
sess = one.alyx.rest('sessions', 'list', django=f'data_dataset_session_related__name__in,{dsets}')
```
>> returns over 18'000 sessions (ephys and training) - the filter acts as an OR, not AND


You cannot chain queries like so:

```
sess = one.alyx.rest('sessions', 'list', django=f'data_dataset_session_related__name__in,_ibl_rightCamera.times.npy,'
                                                f'data_dataset_session_related__name__in,_ibl_bodyCamera.times.npy,'
                                                f'data_dataset_session_related__name__in,_ibl_leftCamera.times.npy')
```
as it searches on the same field.
Error: `"Duplicated fields in \"data_dataset_session_related__name__in,_ibl_rightCamera.times.npy,data_dataset_session_related__name__in,_ibl_bodyCamera.times.npy,data_dataset_session_related__name__in,_ibl_leftCamera.times.npy\"`
